lms/djangoapps/certificates/tests/test_api.py

```diff
@@ -391,6 +398,14 @@ def setUpClass(cls):
             grade="0.99",
             verify_uuid=cls.uuid,
         )
+        # certificate for a course that will be deleted
+        GeneratedCertificateFactory.create(
+            user=cls.student,
+            course_id=cls.course_to_be_deleted.id,
+            status=CertificateStatuses.downloadable
+        )
```

Comment:
Rather than create and delete a course, you can just use a non-existent course_id here.
For examle:
```python
        GeneratedCertificateFactory.create(
            user=cls.student,
            course_id='course-v1:some+fake+course',
            status=CertificateStatuses.downloadable
        )
```

Reply:
That's a very good and cleaner implementation, however, testing this on my devstack makes `test_get_certificate_with_deleted_course` test fail. 
When ran with my implementation runs and executes the test sussesfully. When using this suggested implementation, the function `get_certificate_for_user` returns:
```
AssertionError: course-v1:some+fake+course is not an instance of <class 'opaque_keys.edx.keys.CourseKey'>
```
Should I keep my implementation? Is there any implication in creating and deleting a course (besides test performance)?

Reply:
I've had to use `cls.nonexistent_course_id = CourseKey.from_string('course-v1:some+fake+course')` for it to work.

Thank you for helping me find a solution to this @xitij2000.